File: Arrays-and-hashes/Contains_Duplicate.py
#Question: Contains Duplicate
#Initial approach: double for loop comparission but time complexity goes to O(n^2)
#Solution approach: Use a set and check using one for loop adding to the set if its not there. If it is return true else False
#link: https://leetcode.com/problems/contains-duplicate/submissions/2094176992
#Code:
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seta = set()
        for i in nums:
            if i in seta:
                return True
            else:
                seta.add(i)
        return False
#Time comlexity: O(n)      

File: Arrays-and-hashes/Contains_duplicate.cpp
#Question: Contains Duplicate
#Difficulty: Easy
#Initially: Made coding error on iterator using iterator in the seen condition
#link: https://leetcode.com/problems/contains-duplicate/submissions/2101788168
#Final solution
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        set<int> seen;
        for(auto it=nums.begin();it!=nums.end();it++){
            if(seen.find(*it)!=seen.end()){
                return true;
            }
            else{
                seen.insert((*it));
            }

        }
        return false;
        
    }
};
